dietpi/dietpi-software

```diff
@@ -4170,6 +4177,50 @@ WantedBy=multi-user.target
 _EOF_
 		fi
 
+		if To_Install 141 adsb-feeder # ADSB Feeder
+		then
+			# create the userdata directory and clone the adsb-feeder repo into /tmp
+			G_EXEC mkdir -p /mnt/dietpi_userdata/adsb-feeder
+			G_EXEC_OUTPUT=1 G_EXEC git clone -b dietpi 'https://github.com/dirkhh/adsb-feeder-image.git' /tmp/adsb-feeder
+
+			# remove the service that isn't needed for an app install on DietPi and install the rest
+			G_EXEC cd /tmp/adsb-feeder/src/modules/adsb-feeder/filesystem/root
+			G_EXEC rm -rf ./usr/lib/systemd/system/adsb-bootstrap.service
+			G_EXEC cp -a ./usr/lib/systemd/system/* /etc/systemd/system/
+
+			# determine the version
+			ADSB_FEEDER_DATE_COMPONENT=$(git log -20 --date=format:%y%m%d --format="%ad" | uniq -c | head -1 | awk '{ print $2"."$1 }')
+			ADSB_FEEDER_TAG_COMPONENT=$(git describe --match "v[0-9]*" | cut -d- -f1)
+			ADSB_FEEDER_VERSION="${ADSB_FEEDER_TAG_COMPONENT}(dietpi)-${ADSB_FEEDER_DATE_COMPONENT}"
+
+			# create the target directory for the app and populated with the code from the git checkout
+			G_EXEC mkdir -p /opt
+			G_EXEC cp -a /tmp/adsb-feeder/src/modules/adsb-feeder/filesystem/root/opt/adsb /opt
+			G_EXEC cd /opt/adsb
+
+			# remove the git clone of the repo we installed from
+			G_EXEC rm -rf /tmp/adsb-feeder
+
+			# create a symlink so the config files reside where they should be in /mnt/dietpi_userdata/adsb-feeder
+			G_EXEC mkdir -p /mnt/dietpi_userdata/adsb-feeder/config
+			G_EXEC ln -s /mnt/dietpi_userdata/adsb-feeder/config .
+
+			# set the 'image name' and version that are shown in the footer of the Web UI
+			echo "ADSB Feeder app running on DietPi" > feeder-image.name
+			echo "$ADSB_FEEDER_VERSION" > adsb.im.version
+
+			# get the Python module
+			G_AGI python3-flask
+
+			# finally ensure that /run allows executables (this is needed for the containers to be able to not do excessive
+			# writes to physical storage - which might be an SDcard)
+			G_EXEC mount -o remount,exec /run
+
+			# last but not least, register the services needed with DietPi
+			aENABLE_SERVICES+=('adsb-setup' 'adsb-nonimage' 'adsb-docker')
+			aSTART_SERVICES+=('adsb-setup' 'adsb-nonimage' 'adsb-docker')
```

Comment:
As the services are added to `dietpi-services`, and as such started/stopped automatically on `dietpi-software` executions, they do not need to be defined here.
```suggestion
```
`adsb-nonimage.service` is started via `BintTo` or similar with one of the other two? Else it could be added to `dietpi-services` as well, ordered in the order of which they need to start, in case one depends on the other.

Reply:
the `adsb-nonimage.service` needs to run only once, and it must be guaranteed to run first. I'll look into understanding the interdependencies of these different table some more. I'm still quite new to the codebase and saying that plain bash is not my preferred programming environment would be an understatement 🤣

Reply:
Hmm, currently it runs after `adsb-setup.service`. Since `dietpi-services start` runs before the `aSTART_SERVICES` array, currently `adsb-nonimage.service` would probably need to start right within the install block. If it only needs to run once, is it worth to have it as a service? Or do you mean with "once" that it is a oneshot task (i.e. exits when done) that needs to finish once every time the ADSB feeder is started?

Reply:
it's a service because that way (at least in theory) I can ensure that it runs before adsb-setup (simply with the `Before:` and `After:` tags in the unit file), and because having something as a service makes it much easier to give it a controlled lifecycle. I have seen some of your other comments on just how broken `systemd` is and how much it is not understanding the needs of SBCs... and having worked with some of the people behind `systemd` for 25 years or so, I hear you and I feel you. All that said, it does do a decent job to deal with those kind of tasks.

And yes, the `-nonimage` service really does only run once. I guess I could manually start it right after the install and never enable it... hmm... food for thought

Reply:
Haha, that comment was caused by the recent conversation I had with Lennart regarding this particular dbus/logind topic 😄.

Generally I am very happy with systemd, I love the easiness it adds to service management, sandboxing, ordering/dependencies etc. I wished it was more modular, e.g. that one can install the core init system and service management without e.g. systemd-networkd/resolved which we never use and which Debian itself never uses (only Ubuntu uses it OOTB), but I got used to that since the beginning.

But I get annoyed when multiple services/features which currently can be separately installed (`dbus` is a dedicated non-dependency package) are tied into each other and become mandatory (or loudly complain via error message when not present), even that there are trivially to understand use cases (IMO) which never use those features.

More features are not necessarily bad, and I love the alternatives the systemd project provides for a lot of tasks apart of init/service management, from systemd-boot til systemd-container. But the more features are added, the more important it is to keep things module and optional, IMO, so that not everyone is forced to use the whole package with 90% unused bloat, but can select exactly the needed tools/features from systemd (without the need to recompile it). But yeah, nothing is perfect.